plugins/inputs/aerospike/aerospike.go

```diff
@@ -436,13 +461,75 @@ func parseAerospikeValue(key string, v string) interface{} {
 		return parsed
 	} else if parsed, err := strconv.ParseBool(v); err == nil {
 		return parsed
-	} else if parsed, err := strconv.ParseFloat(v, 32); err == nil {
+	} else if parsed, err := strconv.ParseFloat(v, 64); err == nil {
 		return parsed
 	}
 	// leave as string
 	return v
 }
 
+func (a *Aerospike) parseLatencyInfo(acc telegraf.Accumulator, latencyInfo, hostPort, nodeName string) {
+	latencies := strings.Split(latencyInfo, ";")
+	for i, latency := range latencies {
+		if strings.Compare(latency, "error-no-data-yet-or-back-too-small") == 0 {
+			i++
+		} else {
+			indexOfColon := strings.Index(latency, ":")
+			if indexOfColon == -1 {
+				i++
+				continue
+			}
+			histogram := latency[0:indexOfColon]
+			operation, namespace := splitOperationAndNamespace(histogram)
+			latencyKey := latency[indexOfColon+1:]
+			latencyValue := latencies[i+1]
+			keys := strings.Split(latencyKey, ",")
+			values := strings.Split(latencyValue, ",")
+			nTags := map[string]string{
+				"aerospike_host": hostPort,
+				"node_name":      nodeName,
+				"namespace":      namespace,
+			}
+			nFields := make(map[string]interface{})
+			duration := getDurationOfTransaction(keys[0], values[0])
+			metricKey := operation + "_ops"
+			totalOperations := parseAerospikeValue(metricKey, values[1]).(float64) * duration
+			nFields[metricKey] = totalOperations
+			for i := 2; i < len(keys); i++ {
+				metricKey = operation + strings.Replace(keys[i], ">", "_gt_", -1)
+				nFields[metricKey] = parseAerospikeValue(metricKey, values[i]).(float64) * totalOperations * 0.01
+			}
+			acc.AddFields("aerospike_latency", nFields, nTags, time.Now())
+			i = i + 2
+		}
+	}
+}
+
+func splitOperationAndNamespace(hist string) (operation, namespace string) {
+	if strings.Contains(hist, "{") && strings.Contains(hist, "}") {
+		indexOfOpenCurlyBracket := strings.Index(hist, "{")
+		indexOfCloseCurlyBracket := strings.Index(hist, "}")
+		namespace = hist[indexOfOpenCurlyBracket+1 : indexOfCloseCurlyBracket]
+		operation = hist[indexOfCloseCurlyBracket+2:]
+		return operation, namespace
+	}
+	operation, namespace = hist, "total"
+	return operation, namespace
+}
+
+func getDurationOfTransaction(sTimeWithTimeZone string, eTime string) float64 {
+	timing := strings.Split(sTimeWithTimeZone, "-")
+	sTime := timing[0]
+	startTime, _ := time.Parse(time.TimeOnly, sTime)
+	endTime, _ := time.Parse(time.TimeOnly, eTime)
```

Comment:
Please check the returned errors as otherwise we may get strange results if the data provided by the API changes format or contains invalid values!